### PR TITLE
Create POST API endpoint to ScanCode given GitHub or BitBucket or Git…

### PR DESCRIPTION
**Pull request by RajuKoushik**
… URL.

Fixed issue #52
Signed-off-by: Raju Koushik g.rajukoushik@gmail.com

### CODE REVIEW

File: scanapp/forms.py [hunk 1]
```diff
@@ -30,3 +30,11 @@ class URLScanForm(forms.Form):
 
 class LocalScanForm(forms.Form):
     upload_from_local = forms.FileField(label='Upload from Local')
+
+
+class UrlSpecialScanForm(forms.Form):
```
Comment (pombredanne) [inline scanapp/forms.py:35]: Why create a new form?

File: scanapp/models.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def __str__(self):
 
 class CodeInfo(models.Model):
     def __str__(self):
-        return self.total_code_files
```
Comment (pombredanne) [inline scanapp/models.py:80]: Why do you include model changes in this PR?

File: scanapp/forms.py (hunk 1 above)
Comment (RajuKoushik) [reply scanapp/forms.py:35]: @pombredanne  I had a discussion with @singh1114  where we planned to have a new view for the URL Scan and keep the URLSpecialScan intact for testing purposes which we would be removing after the main view is created.

File: scanapp/views.py [hunk 3]
```diff
@@ -45,6 +45,19 @@
...
+from tasks import scan_code_async_special
+
+from os.path import expanduser
+import os, shutil
+
+import git
+
+
+from urlparse import urlparse
+
```
Comment (pombredanne) [inline scanapp/views.py:60]: Please organize your import correctly

File: scanapp/views.py [hunk 4]
```diff
@@ -45,6 +45,19 @@
...
 from django.views import View
+import subprocess
+from tasks import save_results_to_db
+
+from tasks import scan_code_async_special
+
+from os.path import expanduser
+import os, shutil
+
+import git
```
Comment (pombredanne) [inline scanapp/views.py:56]: What is this? If this is a new dep, add to the requirements

File: scanapp/views.py [hunk 5]
```diff
@@ -151,3 +165,147 @@ def post(self, request):
...
+
+            allowed_exts = ('git')
+            url = urlparse(URL)
+
+            # Create an Instance of InsertIntoDB
+            insert_into_db = InsertIntoDB()
+
+            # call the create_scan_id function
+            scan_id = insert_into_db.create_scan_id(scan_type)
+
```
Comment (pombredanne) [inline scanapp/views.py:191]: All this code about Git should be in two functions:
1. to parse/process the url
2. to fetch.... and fetching SHOULD not be in the view ever... this should be delegated to the background task that fetches and scans

File: scanapp/views.py [hunk 6]
```diff
@@ -151,3 +165,147 @@ def post(self, request):
...
+                else:
+                    path = 'media/user/' + str(request.user) + '/URL/'
+
+                scan_code_async.delay(URL, scan_id, path)
+                # return the response as HttpResponse
+                return HttpResponseRedirect('/resultscan/' + str(scan_id))
+
+
+class UrlScanSpecialView(FormView):
+    template_name = 'scanapp/urlspecialscan.html'
```
Comment (pombredanne) [inline scanapp/views.py:247]: Why would this be a special new view? When we discussed this, I said we would need a single view with a url field and not one per scan

**Review by pombredanne — CHANGES_REQUESTED** [review]
I added some comments. You are missing some basic doc strings for the key functions which should be externalized to be testable separately and there are no tests at all. we need tests.

File: scanapp/views.py (hunk 3 above)
Comment (RajuKoushik) [reply scanapp/views.py:60]: This has been done in the next PR. I'll try to keep this in mind.

File: scanapp/views.py (hunk 6 above)
Comment (RajuKoushik) [reply scanapp/views.py:247]: I had a discussion with @singh1114 where we planned to have a new view for the URL Scan and keep the URLSpecialScan intact for testing purposes which we would be removing after the main view is created.

File: scanapp/views.py (hunk 5 above)
Comment (RajuKoushik) [reply scanapp/views.py:191]: Oh ! I'll add them in 2 different functions and also make sure that fetching is delegated to the background task.

File: scanapp/models.py [hunk 7]
```diff
@@ -77,7 +77,7 @@ def __str__(self):
 
 class CodeInfo(models.Model):
```
Comment (pombredanne) [inline scanapp/models.py:78]: You need to rebase this pr on the latest develop with @singh1114 latest models

File: scanapp/views.py (hunk 5 above)
Comment (pombredanne) [reply scanapp/views.py:191]: @RajuKoushik so why not doing it now?

**Pull request closed without merging**